testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -13038,9 +13050,19 @@ public TesterImpl() {
       super.check(factory, query, typeChecker, parameterChecker, resultChecker);
       final RelDataTypeSystem typeSystem =
           factory.typeSystemTransform.apply(RelDataTypeSystem.DEFAULT);
+      // Create a fake reader so we can invoke the getParser() method.
+      // We don't plan to invoke the parser, so we just pass an empty string.
+      Reader reader = new StringReader("");
+      String parserClass = factory
+          .parserConfig()
+          .parserFactory()
+          .getParser(reader)
+          .getClass()
+          .getCanonicalName();
       final ConnectionFactory connectionFactory =
           factory.connectionFactory
-              .with(CalciteConnectionProperty.TYPE_SYSTEM, uri(FIELD));
+              .with(CalciteConnectionProperty.TYPE_SYSTEM, uri(FIELD))
+              .with(CalciteConnectionProperty.PARSER_FACTORY, parserClass + "#FACTORY");
```

Comment:
We are not propagating any other connection property here so not sure why we should add special handle for the PARSER_FACTORY; it seems a bit inconsistent to pass just this and ignore all other parser properties.

If we want to have some kind of default behavior for the fixture maybe it would be more appropriate to put such logic in `SqlOperatorFixture` and do something similar to `org.apache.calcite.sql.test.SqlOperatorFixture#withConformance` but for the parser factory.